packages/puppeteer-core/src/common/Input.ts

```diff
@@ -439,87 +512,134 @@ export class Mouse {
   }
 
   /**
-   * Dispatches a `mousemove` event.
+   * Moves the mouse to the given coordinate.
+   *
    * @param x - Horizontal position of the mouse.
    * @param y - Vertical position of the mouse.
-   * @param options - Optional object. If specified, the `steps` property
-   * sends intermediate `mousemove` events when set to `1` (default).
+   * @param options - Options to configure behavior.
    */
   async move(
     x: number,
     y: number,
-    options: {steps?: number} = {}
+    options: MouseMoveOptions = {}
   ): Promise<void> {
     const {steps = 1} = options;
-    const fromX = this.#x,
-      fromY = this.#y;
-    this.#x = x;
-    this.#y = y;
+    const points: Point[] = [];
     for (let i = 1; i <= steps; i++) {
+      points.push({
+        x: this.#position.x + (x - this.#position.x) * (i / steps),
+        y: this.#position.y + (y - this.#position.y) * (i / steps),
+      });
+    }
+    await Promise.all(
```

Comment:
why don't we keep it as in the previous version? when the position is updated before all sync calls have been done?

Reply:
We can also rollback the position if the last dispatchMouseCall didn't succeed.

Reply:
I've updated the code so the state management is more transparent.